aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,107 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+  Tensor temp_sol;
+  Tensor temp_lu;
+
+  /* Init to column major format. `sol` and `lu` need to be contiguous
+   * since we pass sol.data() and lu.data() to Lapack */
+  bool tc_sol = isTransposeContiguous(sol);
+  auto self_t = self.view({bx, by}).t();
```

Comment:
I think this breaks if `self` is not viewable to `{bx, by}`.

Reply:
There are only two real blockers:
1. This one.
2. Make the code clearer with restructuring / comment.

Reply:
`self` will always be viewable to `{bx, by}` since they are the dimensions of self. We discussed this earlier, I use `self.view({bx, by})` so that I don't explicitly have to unsqueeze in the case of 1D B (`self`) tensors.

Reply:
Ah yeah.. you are right. My bad. This gets me every time. Can you add a comment here?

Reply:
Sure

Reply:
Is self guaranteed to be contiguous? Otherwise it still cannot be viewed.

Reply:
I basically do this to treat 1D tensors as 2D by viewing them as `{t.size(0), 1}`. If self is 2D it is just the same as viewing it with it's dimensions `{t.size(0), t.size(1)}`.